emu/src/cpu/arm/operations.rs

```diff
@@ -360,6 +318,19 @@ impl Arm7tdmi {
         }
     }
 
+    /// Rotate Right rd by the value in rs, store the result in rd and set condition codes.
+    pub fn ror(&mut self, rd: usize, rs: u32) {
+        let rs = rs & 0xFF;
+        let rd_value = self.registers.register_at(rd);
+        if rs != 0 {
+            let r = shift(ShiftKind::Ror, rs, rd_value, self.cpsr.carry_flag());
+            self.registers.set_register_at(rd, r.result);
+            self.cpsr.set_zero_flag(r.result == 0);
+            self.cpsr.set_sign_flag(r.result.get_bit(31));
```

Comment:
These two we should still set even if the shifted amount is 0

Reply:
Done